src/App.js

```diff
@@ -31,29 +52,46 @@ class App extends Component {
         localStorage.setItem('token', token)
     }
 
+    componentDidMount() {
+        uiStore.subscribe(mergedData => {
+            if (mergedData.type === LOGIN_STATE_CHANGED) {
+                this.setState({
+                    isLoggedIn: mergedData.payload.isLoggedIn
+                })
+            }
+        })
+    }
+
     render() {
+        // when ever there is props ~> there is a route existed
+        const routeProps = this.isRoute(window.location.pathname)
+
+        const redirectToDefault = window.location.pathname === '/' && <Redirect from='/' to='/timeline' />
+        const passToSecureRoute = this.state.isLoggedIn && routeProps && <Route {...routeProps} />
         return (
             <BrowserRouter>
                 <React.Fragment>
                     <Header />
                     <Notifications />
                     <Switch>
-                        <Route path="/timeline" exact component={TimeLine} />
-                        <Route path="/modules" exact component={Modules} />
-                        <Route path="/users" exact component={Users} />
-                        <Route path="/profile" exact component={Profile} />
-
-                        <Route path="/homework" exact
-                            render={props => <ClassPage {...props} studentClass={studentClasses[0]} />} />
-
-                        {studentClasses.map(studentClass => (
-                            <Route key={studentClass} path={"/homework/" + studentClass} exact
-                                render={props => <ClassPage {...props} studentClass={studentClass} />} />
-                        ))}
-                        <Route path="/currentUserProfile" exact component={currentUserProfile} />
-                        <Route path="/userAccount" exact component={userAccount} />
-                        <Route path="/TrainTicket" exact component={TrainTicket} />
-                        <Redirect from="/" to="/timeline" />
+                        <Route path='/timeline' exact component={TimeLine} />
+                        { // Redirect to the timeline if the pathname doesn't matches
+                            redirectToDefault
+                        }
+                        { // Render wich secure route the user did specify
+                            passToSecureRoute
+                        }
+                        {
+                            // For who created this route could you please make it in one line 
```

Comment:
Hi @chileluk Could you please help me with this 🙂

Reply:
@JawharBairakdar What do you need help with?

Reply:
- As you see I am working to make our Routes Not visible for the strangers who are not logged in before.
 - in your clone if you visited the users Route without log in you will see clearly all the users that we have.
 - what am I need a help with in it if you just make your route here more flixible with the `params` or something similer to that; Because I can't secure a route that it's in a map method :), Of course if you have a time for that because it's your addition and I have someother stuff i am working on them at the moment.

Reply:
I've changed this -- it's only rendering the Homework component for the "/homework" route
I've moved the mapping inside the Homework component, so you don't have to worry about that anymore
I'll do a PR soon for that